catalogue_graph/tests/adapters/axiell/test_loader.py

```diff
@@ -0,0 +1,334 @@
+    ) -> list[dict[str, Any]]:
+        self.calls.append((start_time, end_time, max_windows))
+        assert reprocess_successful_windows is False
+        return self.summaries
+
+
+def _request(now: datetime | None = None) -> AxiellAdapterLoaderEvent:
+    now = now or datetime.now(tz=UTC)
+    return AxiellAdapterLoaderEvent(
+        job_id="job-123",
+        window_key="test",
+        window_start=now - timedelta(minutes=15),
+        window_end=now,
+        metadata_prefix="oai",
+        set_spec="collect",
+        max_windows=5,
+    )
+
+
+def _runtime_with(
+    *,
+    store: IcebergWindowStore | None = None,
+    table_client: IcebergTableClient | None = None,
+    oai_client: OAIClient | None = None,
+) -> loader.LoaderRuntime:
+    if table_client is None:
+        # This is a bit of a hack, but we need a table client for the runtime
+        # If tests don't provide one, they probably don't care about it
+        # But we can't easily create a dummy IcebergTableClient without a table
+        # So we'll rely on tests providing it if they need it.
+        # For now, let's assume tests will provide it or we mock it.
+        # But wait, I can't mock it easily without a table.
+        # Let's just require it or let it be None and fail if used?
+        # The LoaderRuntime expects it to be not None.
+        pass
+
+    return loader.LoaderRuntime(
+        store=cast(IcebergWindowStore, store),
+        table_client=cast(IcebergTableClient, table_client),
+        oai_client=cast(OAIClient, oai_client or StubOAIClient()),
+    )
+
+
+def test_execute_loader_updates_iceberg(
+    monkeypatch: pytest.MonkeyPatch,
+    temporary_table: IcebergTable,
+    temporary_window_status_table: IcebergTable,
+) -> None:
+    req = _request()
+    summary = {
+        "window_key": req.window_key,
+        "window_start": req.window_start,
+        "window_end": req.window_end,
+        "state": "success",
+        "attempts": 1,
+        "record_ids": ["id-1"],
+        "last_error": None,
+        "updated_at": req.window_end,
+        "tags": {"job_id": req.job_id, "changeset_id": "changeset-123"},
+        "changeset_id": "changeset-123",
+    }
+    table_client = IcebergTableClient(
+        temporary_table, default_namespace=loader.AXIELL_NAMESPACE
+    )
+    store = IcebergWindowStore(temporary_window_status_table)
+    runtime = _runtime_with(table_client=table_client, store=store)
+
+    def fake_build_harvester(
+        request: AxiellAdapterLoaderEvent,
+        runtime_arg: loader.LoaderRuntime,
+    ) -> StubHarvester:
+        assert request.window_key == req.window_key
+        harvester = StubHarvester([summary])
+        harvester_holder["instance"] = harvester
+        return harvester
+
+    harvester_holder: dict[str, StubHarvester | None] = {"instance": None}
+    monkeypatch.setattr(loader, "_build_harvester", fake_build_harvester)
+
+    response = loader.execute_loader(req, runtime=runtime)
+
```

Comment:
This is how it might look using patch.object

```suggestion
   with patch.object(loader.WindowHarvestManager, 'harvest_recent',
                      return_value=[summary]) as patched_harvest_manager:
        response = loader.execute_loader(req, runtime=runtime)
        patched_harvest_manager.assert_called_once_with(start_time=req.window_start, end_time=req.window_end,
                                                        max_windows=req.max_windows,
                                                        reprocess_successful_windows=False)
```